File: functions/libgcrypt/variants/c/1_ctz_4.c
#include <stdint.h>

int ctz(uint64_t in)
{
 int r = 0;
 if (in == 0) return 64;
 for (uint64_t m = 1; (in & m) == 0; m <<= 1) r++;
 return r;
}


int main(){
    ctz(0);
}

File: functions/libgcrypt/variants/go/1_ctz_4.go
package main

func ctz(in uint64) int32 {
	var r int32
	i := uint(0)
	for (in>>(63-i))&1 == 0 {
		r++
		i++
		if i >= 64 {
			break
		}
	}
	return r
}


func main(){
    ctz(0)
}